packages/flame_bloc/lib/src/flame_bloc_listener.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:bloc/bloc.dart';
+import 'package:flame/components.dart';
+
+import '../flame_bloc.dart';
+
+/// {@template flame_bloc_listener}
+/// A [Component] which can listen to changes in a [Bloc] state.
+/// {@endtemplate}
+class FlameBlocListener<B extends BlocBase<S>, S> extends Component
+    with FlameBlocListenable<B, S> {
+  /// {@macro flame_bloc_listener}
```

Comment:
Is reusing a component doc as THE doc for the constructor a good practice?

Reply:
Generally, the class doc should describe what the class is and how it can be used, whereas the constructor doc answers a more specific question of how the instance is constructed, and what are the various arguments in case it is not clear. Constructor docs are particularly helpful in cases where there is more than one constructor. However, in cases when there is only one constructor and its arguments are obvious, then there doesn't need to be any doc at all: Dart [recommends](https://dart.dev/guides/language/effective-dart/documentation#avoid-redundancy-with-the-surrounding-context) that having a no doc is better than having an obvious or tautological doc.

Another reason to avoid copying class docs for constructors is that a class is a [noun](https://dart.dev/guides/language/effective-dart/documentation#prefer-starting-library-or-type-comments-with-noun-phrases), while a constructor is a [verb](https://dart.dev/guides/language/effective-dart/documentation#prefer-starting-function-or-method-comments-with-third-person-verbs).